kontain-kustomize.sh

```diff
@@ -0,0 +1,151 @@
+            print_help
+        ;;
+        --* | -*)
+            echo "unknown option ${1}"
+            print_help
+        ;; 
+        
+    esac
+    shift
+done
+
+if [ -n  "$tag" ] && [ -n "$location" ]; then 
+    echo "Either release TAG or location of configuration files must be specified" 
+    exit 1
+fi
+
+cloud_provider=$(kubectl get nodes -ojson | jq -r '.items[0] | .spec | .providerID ' | cut -d':' -f1)
+if [ "$cloud_provider" = null ]; then
+    cloud_provider=$(kubectl get node -ojson | jq -r '.items[0] | .metadata | .name')
+fi
+os=$(kubectl get nodes -ojson | jq -r '.items[0] | .status | .nodeInfo | .osImage')
+runtime=$(kubectl get node -ojson | jq -r '.items[0] | .status | .nodeInfo | .containerRuntimeVersion')
+post_process=""
+
+if [ "$cloud_provider" = "azure" ]; then
+    echo "on Azure"
+    overlay=containerd
+elif [ "$cloud_provider" = "aws" ]; then
+    echo "on Amazon"
+    overlay=amazon-eks-custom
+elif [ "$cloud_provider" = "gce" ]; then
+    if [[ $os =~ Ubuntu* ]]; then 
+        echo "on GKE Ubuntu"
+        overlay=containerd
+    elif [[ $os =~ Google ]]; then 
+        echo "On Container-Optimized OS from Google"
+        overlay=gke-gvisor
+    fi
+elif [ "$cloud_provider" = "k3s" ]; then
+    echo "On K3s"
+    overlay=k3s
+    post_process="sudo systemctl restart k3s"
+elif [ "$cloud_provider" = "minikube" ]; then
+    echo "On minikube"
+    if [[ $runtime =~ crio* ]]; then 
+        overlay=crio
+    elif [[ $runtime =~ containerd ]]; then 
+        overlay=containerd
+    else
+        echo "Unsuported runtime"
+    fi
+else
+    echo "If you are runnign K3s make sure change kubectl config by runnig the followinf command:"
+    echo "  export KUBECONFIG=/etc/rancher/k3s/k3s.yaml"
+    echo "If not, you are running Kontain-unsuported cluster provider"
+    exit 1
+fi
+
+if [ -n "$location" ]; then
+    location=${location}/overlays/${overlay}
+else 
+    if [ -z  "$tag" ]; then
+        tags=$(curl -sL https://api.github.com/repos/kontainapp/k8s-deploy/tags | jq  -r '(.[] |select(.name == "current") |.commit|.sha) as $sha | .[] | select(.commit.sha == $sha) | select(.name != "current")|.name')
+
+        for tag in $tags
+        do
+            rel=$(curl -sL https://api.github.com/repos/kontainapp/k8s-deploy/releases/tags/"${tag}" |jq -r 'select(.author.login == "github-actions[bot]") | .id')
+            if [ "$rel" != "null" ]; then 
+                break
+            fi
+        done    
+    fi
+    
+    artifact="https://github.com/kontainapp/k8s-deploy/archive/refs/tags/${tag}.tar.gz"
+    curl -sL "$artifact" | tar -xz --strip-components=1
+    location=./kontain-deploy/overlays/"${overlay}"
+fi
+
+kubectl kustomize "$location" > "${kontain_yaml}"
+
+if [ "$strategy" = "review" ]; then 
+    exit
+fi
+
+kubectl apply --dry-run="$strategy" -f "${kontain_yaml}"
+
+pod=$(kubectl get pods -A -ojson | jq -r '.items[] | .metadata | .name |select(. | startswith("kontain") )')
```

Comment:
what if dry-run?

And if not - what if they aren't there yet?

Reply:
Thank you, good catch